src/OrchardCore/OrchardCore.Search.Elasticsearch.Core/Services/ElasticIndexInitializerService.cs

```diff
@@ -37,30 +38,36 @@ public class ElasticIndexInitializerService : ModularTenantEvents
             _logger = logger;
         }
 
-        public override Task ActivatedAsync()
+        public override async Task ActivatedAsync()
         {
-            if (_shellSettings.IsRunning())
+            if (!_shellSettings.IsRunning())
             {
-                ShellScope.AddDeferredTask(async scope =>
-                {
-                    var elasticIndexSettingsService = scope.ServiceProvider.GetRequiredService<ElasticIndexSettingsService>();
-                    var elasticIndexingService = scope.ServiceProvider.GetRequiredService<ElasticIndexingService>();
-                    var indexManager = scope.ServiceProvider.GetRequiredService<ElasticIndexManager>();
+                return;
+            }
 
-                    var elasticIndexSettings = await elasticIndexSettingsService.GetSettingsAsync();
+            await HttpBackgroundJob.ExecuteAfterEndOfRequestAsync("elastic-initialize", async scope =>
```

Comment:
You're basically moving everything to background task instead of deferred tasks.

Reply:
Yes. Background tasks run in the background without keeping the user waiting until everything is sent to the indexing. Better user experience and ensures that the job finishes